File: CnnNetLib/SequentialActiveInputNeuronGenerator.cs
namespace CnnNetLib
{
    public class SequentialActiveInputNeuronGenerator : IActiveNeuronGenerator
    {
        #region Fields
        
        private readonly int[] _inputNeuronIds;
        private readonly int _activeNeuronCount;

        private int _currentIndex;

        #endregion

        #region Methods

        public int[] GetActiveNeuronIds()
        {
            var ret = new int[_activeNeuronCount];

            for (int i = 0; i < _activeNeuronCount; i++)
            {
                ret[i] = _inputNeuronIds[(_currentIndex + i)%_inputNeuronIds.Length];
            }
            _currentIndex = (_currentIndex + _activeNeuronCount)%_inputNeuronIds.Length;

            return ret;
        }

        #endregion

        #region Instance

        public SequentialActiveInputNeuronGenerator(int[] inputNeuronIds, int activeNeuronCount)
        {
            _inputNeuronIds = inputNeuronIds;
            _activeNeuronCount = activeNeuronCount;
        }

        #endregion
    }
}

File: Java/cnnnetJ/src/SequentialActiveInputNeuronGenerator.java
/**
 * Created with IntelliJ IDEA.
 * User: smiron86
 * Date: 6/11/13
 * Time: 10:35 PM
 */
public class SequentialActiveInputNeuronGenerator implements IActiveNeuronGenerator {

    //region Fields

    private final NeuronInput[] _inputNeurons;
    private final int _activeNeuronCount;

    private int _currentIndex;
    //endregion

    //region Methods
    @Override
    public NeuronInput[] GetActiveNeuronIds() {
        NeuronInput[] ret = new NeuronInput[_activeNeuronCount];

        for (int i = 0; i < _activeNeuronCount; i++) {
            ret[i] = _inputNeurons[(_currentIndex + i) % _inputNeurons.length];
        }
        _currentIndex = (_currentIndex + _activeNeuronCount) % _inputNeurons.length;

        return ret;
    }
    //endregion

    //region Instance
    public SequentialActiveInputNeuronGenerator(NeuronInput[] inputNeurons, int activeNeuronCount) {
        _inputNeurons = inputNeurons;
        _activeNeuronCount = activeNeuronCount;
    }
    //endregion
}
